code/datums/spells/devil.dm

```diff
@@ -0,0 +1,211 @@
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork
+	name = "Summon Pitchfork"
+	desc = "A devil's weapon of choice.  Use this to summon/unsummon your pitchfork."
+	invocation_type = "none"
+	include_user = 1
+	range = -1
+	clothes_req = 0
+	var/obj/item/pitchfork
+	var/pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/
+
+	school = "conjuration"
+	charge_max = 150
+	cooldown_min = 10
+	action_icon_state = "pitchfork"
+	action_background_icon_state = "bg_demon"
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/cast(list/targets, mob/user = usr)
+	if (pitchfork)
+		qdel(pitchfork)
+	else
+		for(var/mob/living/carbon/C in targets)
+			if(C.drop_item())
+				pitchfork = new pitchfork_type
+				C.put_in_hands(pitchfork)
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/Destroy()
+	if(pitchfork)
+		qdel(pitchfork)
+		pitchfork = null
+	return ..()
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/greater
+	pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/greater
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/ascended
+	pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/ascended
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/violin
+	pitchfork_type = /obj/item/device/violin/golden //Oh my god this is the most ghetto code reuse ever.
```

Comment:
If you're fearing this is poor code re-use, you might want to re-class this as a "summon weapon" spell, then move these to subtypes of that